src/applications/vaos/new-appointment/redux/actions.js

```diff
@@ -427,24 +429,37 @@ export function openFacilityPageV2(page, uiSchema, schema) {
   };
 }
 
-export function requestCurrentLocation(uiSchema) {
-  return async dispatch => {
-    dispatch({
-      type: FORM_REQUEST_CURRENT_LOCATION,
-    });
-    try {
-      const location = await getPreciseLocation();
-      dispatch({
-        type: FORM_REQUEST_CURRENT_LOCATION_SUCCEEDED,
-        location,
-        uiSchema,
-      });
-    } catch (e) {
-      captureError(e, false, 'facility page');
+export function updateFacilitySortMethod(sortMethod, uiSchema) {
+  return async (dispatch, getState) => {
+    let location = null;
+    const facilities = getSelectedTypeOfCareFacilities(getState());
+    const calculatedDistanceFromCurrentLocation = facilities.some(
+      f => !!f.legacyVAR?.distancefromCurrentLocation,
+    );
+
+    if (
+      sortMethod === FACILITY_SORT_METHODS.distanceFromCurrentLocation &&
+      !calculatedDistanceFromCurrentLocation
+    ) {
       dispatch({
-        type: FORM_REQUEST_CURRENT_LOCATION_FAILED,
+        type: FORM_REQUEST_CURRENT_LOCATION,
       });
+      try {
+        location = await getPreciseLocation();
+      } catch (e) {
+        captureError(e, false, 'facility page');
+        dispatch({
+          type: FORM_REQUEST_CURRENT_LOCATION_FAILED,
+        });
+      }
     }
+
+    dispatch({
```

Comment:
It seems sort of odd to me that this action fires regardless of whether or not the current location request succeeds. Wouldn't we not be changing the sort method in that case?

Reply:
Updated to only fire if location request was successful, or switching to other sort methods